test/suites/pki.sh

```diff
@@ -106,6 +107,58 @@ test_pki() {
     lxc_remote remote remove pki-lxd
 
 
+    ### Restricted CA signed client certificate with `core.trust_ca_certificates` disabled.
+
+    # Set up the client config
+    cp "${TEST_DIR}/pki/keys/restricted.crt" "${LXD_CONF}/client.crt"
+    cp "${TEST_DIR}/pki/keys/restricted.key" "${LXD_CONF}/client.key"
+    cat "${LXD_CONF}/client.crt" "${LXD_CONF}/client.key" > "${LXD_CONF}/client.pem"
+
+    # Try adding remote using an incorrect token. This should fail even though the client certificate
+    # has been signed by the CA because `core.trust_ca_certificates` is not enabled.
+    ! lxc_remote remote add pki-lxd "${LXD5_ADDR}" --accept-certificate --token=bar || false
+
+    # Add remote using the correct token (restricted).
+    # This should work because the client certificate is signed by the CA.
+    token="$(LXD_DIR=${LXD5_DIR} lxc config trust add --name foo --quiet --restricted)"
+    lxc_remote remote add pki-lxd "${LXD5_ADDR}" --accept-certificate --token "${token}"
+
+    # Should have a trust store entry because `core.trust_ca_certificates` is disabled.
+    lxc_remote config trust ls pki-lxd: | grep restricted
```

Comment:
```suggestion
    lxc_remote config trust ls pki-lxd: | grep -wF restricted
```

To avoid matching on any leftover `unrestricted` cert ;)

Reply:
Done.